### ISSUE TITLE
get_image_data() fails for masked layers

### ISSUE THREAD

**Issue by kocopelly** [EmilDohne/PhotoshopAPI#128]
When trying to get image data from a masked layer, I get 
`ValueError: Invalid array dimension received: { 220220 }. Expected these to sum up to 5,850 but they instead sum up to 48,400. This could be due to the layers' width and height not matching the channel data`

![image](https://github.com/user-attachments/assets/a4c55456-45b1-46d0-bd90-30ad5102676b)

Full traceback:
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[12], [line 5](vscode-notebook-cell:?execution_count=12&line=5)
      [1](vscode-notebook-cell:?execution_count=12&line=1) file_src = psapi.LayeredFile.read('frame.psd') # the psd file with the image to replace
      [3](vscode-notebook-cell:?execution_count=12&line=3) masked_layer = file_src['maskedlayer']
----> [5](vscode-notebook-cell:?execution_count=12&line=5) data = masked_layer.get_image_data()
      [7](vscode-notebook-cell:?execution_count=12&line=7) data

ValueError: Invalid array dimension received: { 220220 }. Expected these to sum up to 5,850 but they instead sum up to 48,400. This could be due to the layers' width and height not matching the channel data
```

Reproduceable:
```
file_src = psapi.LayeredFile.read('frame.psd') # the psd file with the image to replace

masked_layer = file_src['maskedlayer']

data = masked_layer.get_image_data()

data
```

Note: If I put the layer in a group, and mask the group, this error does not happen.

**Comment by EmilDohne** [comment]
Hmm yeah I think I know why this might be happening but I need to investigate further.

In short, photoshop optimizes away masks quite aggressively, so rather than storing the full mask it only stores the tight bounding box of it instead of the layers bounding box (which could be smaller or larger). It could be that during some of my recent refactors this was not accounted for. Hopefully should be able to find a fix for this tomorrow. Thanks for the flag!

**Comment by EmilDohne** [comment]
Hey @kocopelly sorry for only getting around to this now. This should be fixed now and the release CI for the v0.5.2 is running so once that is out you should be able to just pip install the new version from there :)

**Comment by EmilDohne** [comment]
This is now out and you can install it with pip!
